Mathlib/AlgebraicGeometry/Morphisms/Separated.lean

```diff
@@ -113,6 +114,83 @@ instance [IsSeparated g] :
 
 end IsSeparated
 
+section of_injective
+
+open Scheme Pullback
+
+variable (𝒰 : Y.OpenCover) (𝒱 : ∀ i, (pullback f (𝒰.map i)).OpenCover)
+
+lemma Scheme.Pullback.diagonalCoverDiagonal_eq_top_of_injective (hf : Function.Injective f.base) :
+    diagonalCoverDiagonal f 𝒰 𝒱 = ⊤ := by
+  rw [← top_le_iff]
+  rintro x -
+  simp only [diagonalCoverDiagonal, openCoverOfBase_J, openCoverOfBase_obj, openCoverOfLeftRight_J,
+    Opens.iSup_mk, Opens.carrier_eq_coe, Hom.opensRange_coe, Opens.coe_mk, Set.mem_iUnion,
+    Set.mem_range, Sigma.exists]
+  have H : (pullback.fst f f).base x = (pullback.snd f f).base x :=
+    hf (by rw [← Scheme.comp_base_apply, ← Scheme.comp_base_apply, pullback.condition])
+  let i := 𝒰.f (f.base ((pullback.fst f f).base x))
+  obtain ⟨y : 𝒰.obj i, hy : (𝒰.map i).base y = f.base _⟩ :=
+    𝒰.covers (f.base ((pullback.fst f f).base x))
+  obtain ⟨z, hz₁, hz₂⟩ := exists_preimage_pullback _ _ hy.symm
+  let j := (𝒱 i).f z
+  obtain ⟨w : (𝒱 i).obj j, hy : ((𝒱 i).map j).base w = z⟩ := (𝒱 i).covers z
+  refine ⟨i, j, ?_⟩
+  simp_rw [diagonalCover_map]
+  show x ∈ Set.range _
+  erw [range_map]
```

Comment:
```suggestion
  simp_rw [diagonalCover_map, ← Set.mem_range]
  erw [range_map]
```
I am wondering why this `erw` is necessary. I did not manage to fix it, but I made the observation that it has to be related to the `x ∈`, since applying `range_map` without it works with `rw`.